src/d3a/d3a_core/util.py

```diff
@@ -330,3 +334,20 @@ def recursive_retry(functor, retry_count, max_retries, *args, **kwargs):
         if retry_count >= max_retries:
             raise e
         return recursive_retry(functor, retry_count+1, max_retries, *args, **kwargs)
+
+
+def validate_const_settings_for_simulation():
+    from d3a.models.const import ConstSettings
+    # If schemes are not compared and an individual scheme is selected
+    # And the market type is not single sided market
+    # This is a wrong configuration and an exception is raised
+    if not ConstSettings.IAASettings.AlternativePricing.COMPARE_PRICING_SCHEMES and \
+       ConstSettings.IAASettings.MARKET_TYPE != 1 and \
+       ConstSettings.IAASettings.AlternativePricing.PRICING_SCHEME != 0:
+        assert False, "Alternate pricing schemes are only usable with an one sided market."
+
+    # If an alternate price is selected on compare schemes
+    # There should be a single sided market
+    if ConstSettings.IAASettings.AlternativePricing.COMPARE_PRICING_SCHEMES and \
```

Comment:
This condition should never be True, as the user sets 
either
COMPARE_PRICING_SCHEMES 
or
PRICING_SCHEME
So I would change the `and` into an `or`. Do you agree?

Reply:
Additionally, how about adding an assert here for the validation of COMPARE_PRICING_SCHEMES and PRICING_SCHEME ? We first decided against it, but as you added this function, it would be beneficial for the user , IMO.

Reply:
To answer both comments, the problem is not with the user in this case, but with the subprocesses. When COMPARE_PRICING_SCHEMES is set to true, simulation.py will change the PRICING_SCHEME according to the subprocess that is running. Therefore it is a valid scenario in our case, and we should not assert if we encounter it, because otherwise some processes would fail.

Reply:
OK, thanks, I understand